src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -585,12 +585,12 @@ class spell_dk_death_strike : public SpellScriptLoader
             void HandleHeal(SpellEffIndex /*effIndex*/)
             {
                 Unit* caster = GetCaster();
-                //TODO: heal = std::min(10% health, 20% of all damage taken in last 5 seconds)
-                int32 heal = CalculatePct(caster->GetMaxHealth(), GetSpellInfo()->GetEffect(EFFECT_4)->CalcValue());
-                caster->CastCustomSpell(SPELL_DK_DEATH_STRIKE_HEAL, SPELLVALUE_BASE_POINT0, heal, caster, true);
+                // heal for 25% taken for last 5 seconds, for min 7% of max health, up to 
```

Comment:
Comment content improvement suggestion:
```suggestion
                // Heal 25% damage taken during the last 5 seconds, up to a minimum of 7% of max health.
```
(in case I have misunderstood the spell description, please adjust the text accordingly)

Reply:
actually I don't think the usage of "up to" with "minimum" is a good one in genereal (which is also what I did). I'll simply use the actual tooltip text:
"heals you for 25% of all damage taken in the last 5 sec, minimum 7.0% of maximum health."

Reply:
Fair & good point. So maybe the following will be acceptable?
```suggestion
                // Heals you for 25% of all damage taken in the last 5 sec, minimum 7.0% of maximum health.
```